src/frontend.md

```diff
@@ -499,6 +499,83 @@ export function fakeTicket() {}
    const documentMutation = usePatchDocument(documentId);
    ```
 
+## I18n
+
+### Translation keys
+
+1. Use `snake_case` for translation keys.
+
+   ```typescript
+   // bad
+   t('home.helloWorld');
+
+   // good
+   t('home.hello_world');
+   ```
+
+1. Do not split phrases into several translation keys. Sometimes it's impossible to translate splitted phrase to another language.
+
+   ```typescript
+   // bad
+   const greeting = t('home.hello') + userName + '!';
+
+   // good
+   const greeting = t('home.hello_user', { userName });
+   ```
+
+1. Give meaningful names to translation keys. Translation key should shortly describe their purpose of content.
+
+   ```typescript
+   // bad
+   t('home.header_title');
```

Comment:
bad and good are mixed up by an error or you really mean it?

Reply:
taken from here: https://www.notion.so/osome/i18n-cf5a8066958b42bcb2748b632b4ab9d1 key naming point 3

Reply:
But `header_title` looks exactly as "purpose of content" :)

Reply:
OK, there are 2 different rules which balance each other: 1) don't translate literally `You have not created a ticket yet?` to `blank_state_message` and 2) add some context, so use `chose_locale_dropdown` instead of just `dropdown`

Reply:
let's add 2 separate rules close to each other:

1. Use clear and meaningful key names that succinctly describe their purpose and the value they hold. This is similar to how variables are named in code.

bad: button
good: create_ticket_button

2. Avoid being overly specific and refrain from using translation key values as names. 

bad: you_have_not_created_any_ticket_yet
good: blank_state_text